### PR TITLE
net/gnrc_netif: add support for priority queues

### PR DESCRIPTION
**Pull request by jia200x**
### Contribution description
This PR adds support for `gnrc_netif` priority queues.
Since some drivers run a Bottom Half Processor in the same `gnrc_netif` thread, events from MAC layers (e.g submac) can delay IRQ offloading. To avoid that, a second event queue with high priority is introduced if `MODULE_BHP_EVENT` is present.

I also use this queue for the event based Bottom Half Processor of `kw2xrf` radios.

### Testing procedure
Make sure all unittests pass. Also, `kw2xrf` should still work in GNRC.
Since I'm currently on the train I couldn't test it on real hardware. I will post tests results asap.

### Issues/PRs references
This should make MAC layers such as the SubMAC or openDSME more robust.

### CODE REVIEW

File: sys/net/gnrc/netif/init_devs/auto_init_kw2xrf.c [hunk 1]
```diff
@@ -56,11 +56,11 @@ void auto_init_kw2xrf(void)
         LOG_DEBUG("[auto_init_netif] initializing kw2xrf #%u\n", i);
 
         /* Init Bottom Half Processor (with events module) and radio */
-        bhp_event_init(&kw2xrf_bhp[i], &_netif[i].evq, &kw2xrf_radio_hal_irq_handler, &kw2xrf_netdev[i].submac.dev);
+        bhp_event_init(&kw2xrf_bhp[i], &_netif[i].evq[GNRC_NETIF_EVQ_INDEX_PRIO_HIGH],
+                       &kw2xrf_radio_hal_irq_handler, &kw2xrf_netdev[i].submac.dev);
         kw2xrf_init(&kw2xrf_devs[i], (kw2xrf_params_t*) p,&kw2xrf_netdev[i].submac.dev,
```
Comment (benpicco) [inline sys/net/gnrc/netif/init_devs/auto_init_kw2xrf.c:61]:
```suggestion
        kw2xrf_init(&kw2xrf_devs[i], p, &kw2xrf_netdev[i].submac.dev,
```

**Review by benpicco — APPROVED** [review]
Looks sensible to me

**Comment by jia200x** [comment]
I already tested this one with https://github.com/RIOT-OS/RIOT/pull/18472 and it definitely helps the SubMAC or any MAC layer in the middle

**Comment by benpicco** [comment]
CI is unhappy

File: sys/net/gnrc/netif/gnrc_netif.c [hunk 2]
```diff
@@ -1697,8 +1709,7 @@ static void _process_events_await_msg(gnrc_netif_t *netif, msg_t *msg)
         event_t *evp;
         /* We can not use event_loop() or event_wait() because then we would not
          * wake up when a message arrives */
-        event_queue_t *evq = &netif->evq;
-        while ((evp = event_get(evq))) {
+        while ((evp = _gnrc_netif_fetch_event(netif))) {
```
Comment (benpicco) [inline sys/net/gnrc/netif/gnrc_netif.c:1712]: Other users of `gnrc_netif_t` now need an update too

**Comment by jia200x** [comment]
I forgot GNRC LoRaWAN and the other MACs use `event_post` too... Just amended and squashed directly.

**Comment by jia200x** [comment]
Thanks for the review!

**Pull request merged**